integration_tests/suite/test_user_room.py

```diff
@@ -116,6 +117,122 @@ def test_list_by_user_uuid(self, room_1, room_2, room_3):
+        )
+        # user2 posts a message in room3
+        self.chatd.rooms.create_message_from_user(
+            room_uuid=room3['uuid'],
+            message_args={'content': 'test message 4'},
+        )
+
+        # user2 can see messages from room2
+        messages = self.chatd.rooms.list_messages_from_user(room2['uuid'])
+        assert_that(
+            messages,
+            has_entries(
+                items=contains_inanyorder(
+                    has_entries(
+                        content='test message 3', room=has_entries(uuid=room2['uuid'])
+                    ),
+                ),
+                total=equal_to(1),
+                filtered=equal_to(1),
+            ),
+        )
+
+        # user2 can see messages from room3
+        messages = self.chatd.rooms.list_messages_from_user(room3['uuid'])
+        assert_that(
+            messages,
+            has_entries(
+                items=contains_inanyorder(
+                    has_entries(
+                        content='test message 2', room=has_entries(uuid=room3['uuid'])
+                    ),
+                    has_entries(
+                        content='test message 4', room=has_entries(uuid=room3['uuid'])
+                    ),
+                ),
+                total=equal_to(2),
+                filtered=equal_to(2),
+            ),
+        )
+
+        # user2 listing messages from room1 raises 404 not found
+        assert_that(
+            calling(self.chatd.rooms.list_messages_from_user).with_args(
+                room_uuid=room1['uuid'], token=user2_token
+            ),
+            raises(HTTPError, has_properties(status_code=404)),
+        )
+
+        self.chatd.set_token(user1_token)
+        # user1 can see messages from room1
+        messages = self.chatd.rooms.list_messages_from_user(room1['uuid'])
+        assert_that(
+            messages,
+            has_entries(
+                items=contains_inanyorder(
+                    has_entries(
+                        content='test message 1', room=has_entries(uuid=room1['uuid'])
+                    ),
+                ),
+                total=equal_to(1),
+                filtered=equal_to(1),
+            ),
+        )
+
+        # user1 can see messages from room3
+        messages = self.chatd.rooms.list_messages_from_user(room3['uuid'])
+        assert_that(
+            messages,
+            has_entries(
+                items=contains_inanyorder(
+                    has_entries(
+                        content='test message 2', room=has_entries(uuid=room3['uuid'])
+                    ),
+                    has_entries(
+                        content='test message 4', room=has_entries(uuid=room3['uuid'])
+                    ),
+                ),
+                total=equal_to(2),
+                filtered=equal_to(2),
+            ),
+        )
+
+        # user1 listing messages from room2 raises 404 not found
```

Comment:
Most comments were not very useful. If the code is not clear, please split into multi tests and refactor/rename method/variable